File: ros/hbba_lite/scripts/hbba_lite_python_tests/test_on_off_hbba_publisher.py
#!/usr/bin/env python3
import rospy

from std_msgs.msg import Int8

import hbba_lite


def main():
    rospy.init_node('test_on_off_hbba_publisher')
    pub = hbba_lite.OnOffHbbaPublisher('int_topic', Int8, queue_size=10)

    rate = rospy.Rate(1) # 1hz
    i = 0
    while not rospy.is_shutdown():
        pub.publish(Int8(i))
        i += 1
        rate.sleep()


if __name__ == '__main__':
    main()

File: ros/hbba_lite/src/hbba_lite_cpp_tests/test_on_off_hbba_publisher.cpp
#include <ros/ros.h>
#include <std_msgs/Int8.h>

#include <hbba_lite/Publishers.h>

int main(int argc, char** argv)
{
    ros::init(argc, argv, "test_on_off_hbba_publisher");
    ros::NodeHandle nodeHandle;

    OnOffHbbaPublisher<std_msgs::Int8> pub(nodeHandle, "int_topic", 10);

    ros::Rate rate(1);
    for (int i = 0; ros::ok(); i++)
    {
        std_msgs::Int8 msg;
        msg.data = i;
        pub.publish(msg);

        ros::spinOnce();
        rate.sleep();
    }

    return 0;
}
